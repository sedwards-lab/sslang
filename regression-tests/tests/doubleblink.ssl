toggle(led : Ref (Sched Int)) =
  led <- 1 - led

slow(led : Ref (Sched Int)) =
  let e1 = Occur : Sched Event
  loop
    toggle led
    30 ms later e1 <- Occur
    wait e1

fast(led : Ref (Sched Int)) =
  let e2 = Occur : Sched Event
  loop
    toggle led
    20 ms later e2 <- Occur
    wait e2
    
main(led : Ref (Sched Int)) =
  par slow led
      fast led