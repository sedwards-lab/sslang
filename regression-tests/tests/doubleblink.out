    20000000 led: 1
    30000000 led: 0
    40000000 led: 1
    60000000 led: 1
    80000000 led: 0
    90000000 led: 1
   100000000 led: 0
   120000000 led: 0
   140000000 led: 1
   150000000 led: 0
   160000000 led: 1
   180000000 led: 1
   200000000 led: 0
   210000000 led: 1
   220000000 led: 0
   240000000 led: 0
   260000000 led: 1
   270000000 led: 0
   280000000 led: 1
   300000000 led: 1
   320000000 led: 0
   330000000 led: 1
   340000000 led: 0
   360000000 led: 0
   380000000 led: 1
   390000000 led: 0
   400000000 led: 1
   420000000 led: 1
   440000000 led: 0
   450000000 led: 1
   460000000 led: 0
   480000000 led: 0
   500000000 led: 1
